httpreplay/httpreplay.go

```diff
@@ -46,10 +47,14 @@ type Recorder struct {
 
 // NewRecorder creates a recorder that writes to filename. The file will
 // also store initial state that can be retrieved to configure replay.
+// The Recorder MITM proxy shall be configured with a custom certificate by
+// providing paths to cert and key, in case where the custom certificate is not
+// provided, the proxy will be created with an auto-generated certificate.
+//
 //
 // You must call Close on the Recorder to ensure that all data is written.
-func NewRecorder(filename string, initial []byte) (*Recorder, error) {
-	p, err := proxy.ForRecording(filename, 0)
+func NewRecorder(filename string, initial []byte, cert, key string) (*Recorder, error) {
```

Comment:
Since many people won't need these, and to avoid changing the API, can we add a method to Recorder to set these instead?

```
n := httpreplay.NewRecorder(filename, nil)
n.SetCert(cert, key)
...
```

Reply:
gee that makes good sense, sorry I was tunnel visioned. I will make that change.

Reply:
Hi @jba, would you mind please having a look again?